### PR TITLE
add clear call history feature

### LINKED ISSUE #5: Cleanup of recent / missed calls necessary
I consider all aspects of not deleting data in a structured way as a bug in information systems.

That said, the dialer app does not clean up the list of recent or missed calls and neither has an option for the user to do it. This is really necessary, my list goes back 2 years now, and its getting slower and slower.

### PR DESCRIPTION
**Pull request by lduboeuf**
This will allow to clear the call history from the dialer-app.

Pre-requisite: https://github.com/ubports/history-service/pull/40

fixes: #5 

Action available at the right corner of the history screen.
screenshot20210630_143534599

Confirmation popup :
screenshot20210708_170320385

Wait screen:
screenshot20210630_144112870

For testing; Make sure to backup history database , e.g from your PC: `adb pull /home/phablet/.local/share/history-service/history.sqlite history.sqlite` 
install PR-40 from history-service: `sudo ubports-qa install history-service 40`

Dialer-app click package for test:
see https://github.com/lduboeuf/dialer-app/actions , check for last build and choose artefact below

To restore: `adb push history.sqlite /home/phablet/.local/share/history-service/history.sqlite ` , reboot your device

### CODE REVIEW

**Comment by Flohack74** [comment]
Finally! :)

File: src/qml/HistoryPage/HistoryCleaner.qml
Comment (UniversalSuperBox) [inline src/qml/HistoryPage/HistoryCleaner.qml]: Missed a translation

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]

**Comment by lduboeuf** [comment]
@UniversalSuperBox or @all how to deal with that issue of exposing a "delete all" feature from History QML component ?

We already can do it with some js, btw. ( not a viable solution, too heavy operation )

**Comment by UniversalSuperBox** [comment]
@lduboeuf I would expect a `DROP TABLE` followed by setting it up again would be enough in this case, and would be much faster than transactionally deleting every record.

**Comment by lduboeuf** [comment]
> @lduboeuf I would expect a `DROP TABLE` followed by setting it up again would be enough in this case, and would be much faster than transactionally deleting every record.

Humm, of course that would be easier and quicker. So you mean no need to provide delete after N months ?
So about my former question, you finally don't see any security issue exposing the "deleteAll" to HistoryManager QML component ?

**Comment by UniversalSuperBox** [comment]
You asked about deleting *all* history, not a range :P

I'm pretty sure that history-service is unused by anything but messaging-app and dialer-app. If that changes in the future then we should consider locking it down.

**Comment by lduboeuf** [comment]
ok, my question was not clear sorry.

**Comment by lduboeuf** [comment]
So to keep that simpler, i could only provide a "deleteAll" method instead of dealing with date range. That would make the history-service PR lighter. I will give a try

**Comment by UniversalSuperBox** [comment]
I would still try to design a way to delete history past a certain point, rather than the entire history. Deleting the whole history is potentially useful, deleting a part of it more so. It's just a matter of getting the service in order to do such a thing. It might not be easy.

**Comment by lduboeuf** [comment]
right, ok. So lets keep it like this for now, i will correct the requested change.

**Comment by Flohack74** [comment]
@lduboeuf ready to merge?

**Comment by lduboeuf** [comment]
> @lduboeuf ready to merge?

yes

**Comment by lduboeuf** [comment]
@Flohack74 see related PR in history-service. Review is still pending it seems, so waiting for approval

**Comment by Flohack74** [comment]
Got it, will ping ;)

File: src/qml/HistoryPage/HistoryCleaner.qml
Comment (Flohack74) [inline src/qml/HistoryPage/HistoryCleaner.qml:29]: Please check also here if pot file changes are needed :)

**Review by Flohack74 — APPROVED** [review]

**Pull request merged**
